src/server/pps/server/api_server.go

```diff
@@ -1243,7 +1248,70 @@ func (a *apiServer) collectDatums(ctx context.Context, job *pps.Job, cb func(*da
 }
 
 func (a *apiServer) CreateDatum(server pps.API_CreateDatumServer) (retErr error) {
-	return status.Errorf(codes.Unimplemented, "method CreateDatum not implemented")
+	msg, err := server.Recv()
+	if err != nil {
+		return errors.Wrap(err, "server receive")
+	}
+	if msg.Input == nil {
+		return errors.Errorf("first message must specify an input")
+	}
+
+	it, err := a.getStreamingIterator(server.Context(), msg.Input)
+	if err != nil {
+		return errors.Wrap(err, "getting streaming iterator")
+	}
+	for {
+		number := msg.Number
```

Comment:
I noticed this while typing the above comment, what happens if number is -2?  The conditional below checks == 0, so you will get 2^64-3 datums ;)

Reply:
Yea, good callout. I was considering allowing a negative `number` to indicate to return all datums, but I'm leaning towards not doing that. If a user wants all datums, they should just use `ListDatum`.